### PR TITLE
fix: build-recommendation-object.yml workflow

### PR DESCRIPTION
**Pull request by kpoineal**
# Overview/Summary

Updated Recommendation Object creation to be done as part of PR validation and included in new changes.

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: .github/workflows/build-recommendation-object.yml [hunk 1]
```diff
@@ -1,62 +1,41 @@
...
+      - 'azure-resources/**/*.kql'
+      - 'azure-specialized-workloads/**/*.yaml'
+      - 'azure-specialized-workloads/**/*.kql'
+      - 'azure-waf/**/*.yaml'
+      - 'azure-waf/**/*.kql'
+  workflow_dispatch:
 
 permissions:
-  contents: read
+    contents: write
```
Comment (oZakari) [inline .github/workflows/build-recommendation-object.yml:16]:
```suggestion
    contents: read
```

Comment (oZakari) [reply .github/workflows/build-recommendation-object.yml:16]: Scope permissions at job level

File: .github/workflows/build-recommendation-object.yml [hunk 2]
```diff
@@ -1,62 +1,41 @@
...
 
 jobs:
-  build:
```
Comment (oZakari) [inline .github/workflows/build-recommendation-object.yml:12]: update-json-object

File: .github/workflows/build-recommendation-object.yml [hunk 3]
```diff
@@ -1,62 +1,41 @@
...
-      - name: Merge PR
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          pr_number=$(gh pr list --state open --limit 1 --json number --jq '.[0].number')
-          gh pr merge $pr_number --merge
-        shell: bash
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
```
Comment (oZakari) [inline .github/workflows/build-recommendation-object.yml:24]:
```suggestion
      uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
```

Comment (oZakari) [reply .github/workflows/build-recommendation-object.yml:24]: Best practice to use commit sha of action. Also v4 is latest.

**Review by oZakari — CHANGES_REQUESTED** [review]

File: .github/workflows/build-recommendation-object.yml [hunk 4]
```diff
@@ -1,62 +1,41 @@
...
+      - 'azure-waf/**/*.kql'
+  workflow_dispatch:
 
 permissions:
-  contents: read
+    contents: write
 
 jobs:
-  build:
+  Generate-Documentation:
```
Comment (oZakari) [inline .github/workflows/build-recommendation-object.yml:19]:
```suggestion
  update-json-object:
```

File: .github/workflows/build-recommendation-object.yml [hunk 5]
```diff
@@ -1,62 +1,41 @@
...
+      - 'azure-resources/**/*.kql'
+      - 'azure-specialized-workloads/**/*.yaml'
+      - 'azure-specialized-workloads/**/*.kql'
+      - 'azure-waf/**/*.yaml'
+      - 'azure-waf/**/*.kql'
+  workflow_dispatch:
 
 permissions:
-  contents: read
+    contents: read
```
Comment (oZakari) [inline .github/workflows/build-recommendation-object.yml:16]:
```suggestion
   contents: read
```

File: .github/workflows/build-recommendation-object.yml [hunk 6]
```diff
@@ -1,62 +1,41 @@
...
+      - 'azure-resources/**/*.kql'
+      - 'azure-specialized-workloads/**/*.yaml'
+      - 'azure-specialized-workloads/**/*.kql'
+      - 'azure-waf/**/*.yaml'
+      - 'azure-waf/**/*.kql'
+  workflow_dispatch:
 
 permissions:
-  contents: read
+   contents: read
```
Comment (oZakari) [inline .github/workflows/build-recommendation-object.yml:16]:
```suggestion
  contents: read
```

**Review by oZakari — APPROVED** [review]

**Review by ejhenry — APPROVED** [review]

**Pull request merged**